rascal2/widgets/controls/model.py

```diff
@@ -0,0 +1,42 @@
+import warnings
+
+from pydantic import ValidationError
+from RATapi.controls import fields
+
+from rascal2.core.commands import editControls
+
+
+class FitSettingsModel:
+    """TableModel to interface Controls fit settings with the view."""
+
+    def __init__(self, presenter):
+        super().__init__()
+        self.presenter = presenter
+        self.controls = self.presenter.model.controls
+        self.model_fields = self.controls.model_fields
+        self.undo_stack = self.presenter.undo_stack
+        self.editable = True
+        self.last_validation_error = None
+
+    def data(self, setting):
+        value = getattr(self.controls, setting)
+        return value
+
+    def setData(self, setting, value) -> bool:
+        # we have to check validation in advance because PyQt doesn't return
```

Comment:
I think there is a fix for exception falling over in C++. I will set it up when I do the logging issue.

As for bad changes in the stack it is possible to mark a command as obsolete https://doc.qt.io/qt-6/qundocommand.html#setObsolete

So we might be able to refactor this later on